tracer/test/Datadog.Trace.Tools.Analyzers.Tests/PublicApiAnalyzerTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿// <copyright file="PublicApiAnalyzerTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Xunit;
+using Verifier = Microsoft.CodeAnalysis.CSharp.Testing.XUnit.AnalyzerVerifier<
+    Datadog.Trace.Tools.Analyzers.PublicApiAnalyzer.PublicApiAnalyzer>;
+
+namespace Datadog.Trace.Tools.Analyzers.Tests
+{
+    public class PublicApiAnalyzerTests
+    {
+        private const string DiagnosticId = PublicApiAnalyzer.PublicApiAnalyzer.DiagnosticId;
+
+        public static string[] GetPublicApiAttributes { get; } = { "PublicApi", "PublicApiAttribute" };
```

Comment:
I would suggest to add two more:
`Fully.Qualified.Name.PublicApi` and `Fully.Qualified.Name.PublicApiAttribute`.

Reply:
I don't think we need these, as they're a subset of the two attributes already added 🤔 There's an argument that we should use those _instead_ but I'm not sure that's worth worrying about right now, until we define a _second_ `[PublicApi]` attribute. Also, depending only on the attribute name makes things quite a lot simpler I think, so I'd prefer that approach until we need something different 😄

Reply:
Just as a clarification, my comment was about making sure that we capture these scenarios:
```
[XXX.PublicApi]
private string _field;
```
Not sure it's worthwhile doing, but it might help us verify we capture those scenarios as the analyzer is looking for the class name:
https://github.com/DataDog/dd-trace-dotnet/blob/9d97e093942674e3dc480e5248e6a07da6e9002e/tracer/src/Datadog.Trace.Tools.Analyzers/PublicApiAnalyzer/PublicApiAnalyzer.cs#L295
Feel free to resolve.

Reply:
Ooooh, I see! That makes sense and should be an easy fix, I'll add it now. The only case that wouldn't work is if you're using an alias, but I think that's fine tbh.

Reply:
Good point. Also child classes of `PublicApiAttribute` will not be captured but I guess that's also something we can tolerate.

Reply:
Yeah, that's easy enough 👍

Reply:
> Not sure it's worthwhile doing, but it might help us verify that we pay attention to the class name itself

Done in https://github.com/DataDog/dd-trace-dotnet/pull/3704/commits/d68cbb468095dc4fd7edd9390f26741ea43d9480

> Can we make PublicApiAttribute sealed

Done in https://github.com/DataDog/dd-trace-dotnet/pull/3704/commits/3b6d752cda098addaefb982c4dd74344e6f3147a

Reply:
Resolving.